File: src/dataset/python/prime-factors_exercism-ThomasZumsteg/openAI/ChatGPT4_prime_factors_v3.py
"""Finds the prime factors of a number"""

def prime_factors(number):
    """Finds prime factors of a number"""
    factors = []
    factor = 2
    while factor * factor <= number:
        while number % factor == 0:
            factors.append(factor)
            number //= factor
        factor += 1
    if number > 1:
        factors.append(number)
    return factors

File: src/dataset/javascript/prime-factors_exercism-elenamountz/openAI/ChatGPT4_prime-factors_v3.js
export const primeFactors = (number) => {
  const factors = [];
  for (let divisor = 2; divisor * divisor <= number; divisor++) {
    while (number % divisor === 0) {
      factors.push(divisor);
      number /= divisor;
    }
  }
  if (number > 1) factors.push(number);
  return factors;
};